tests/src/Traits/DrsIO.php

```diff
@@ -0,0 +1,221 @@
+            );
+      }
+
+        return $answer;
+    });
+    return $this->doAsk($question);
+  }
+
+  /**
+   * Writes an array to the screen as a formatted table.
+   *
+   * @param string[] $array
+   *   The unformatted array.
+   * @param string[] $headers
+   *   The headers for the array. Defaults to ['Property','Value'].
+   */
+  protected function printArrayAsTable(
+        array $array,
+        array $headers = ['Property', 'Value']
+    ): void {
+    $table = new Table($this->output);
+    $table->setHeaders($headers)
+            ->setRows(ArrayManipulator::convertArrayToFlatTextArray($array))
+            ->render();
+  }
+
+  /**
+   * Writes a particular configuration key's value to the log.
+   *
+   * @param string[] $array
+   *   The configuration.
+   * @param string $prefix
+   *   A prefix to add to each row in the configuration.
+   * @param int $verbosity
+   *   The verbosity level at which to display the logged message.
+   */
+  protected function logConfig(array $array, string $prefix = '', int $verbosity = OutputInterface::VERBOSITY_VERY_VERBOSE): void {
+    if ($this->getOutput()->getVerbosity() >= $verbosity) {
+      if ($prefix) {
+        $this->getOutput()->writeln("<comment>Configuration for $prefix:</comment>");
+        foreach ($array as $key => $value) {
+          $array["$prefix.$key"] = $value;
+          unset($array[$key]);
+        }
+      }
+      $this->printArrayAsTable($array);
+    }
+  }
+
+  /**
+   * Asks a required question.
+   *
+   * @param string $text
+   *   Prints the output text.
+   */
+  protected function writeln($text): void {
+    $this->getOutput()->writeln($text);
+  }
+
+  /**
+   * Formats the output message to terminal.
+   *
+   * @param string $text
+   *   Output message to format.
+   * @param int $length
+   *   Length of the output.
+   * @param string $format
+   *   Given string format.
+   */
+  protected function formattedOutput($text, $length, $format): void {
+    $lines = explode("\n", trim($text, "\n"));
+    $maxLineLength = array_reduce(array_map('strlen', $lines), 'max');
+    $length = max($length, $maxLineLength);
+    $len = $length + 2;
+    $space = str_repeat(' ', $len);
+    $this->writeln(sprintf($format, $space));
+    foreach ($lines as $line) {
+      $line = str_pad($line, $length, ' ', STR_PAD_BOTH);
+      $this->writeln(sprintf($format, " $line "));
+    }
+    $this->writeln(sprintf($format, $space));
+  }
+
+  /**
+   *
+   * @return $this
+   *
+   * @see \Robo\Contract\OutputAwareInterface::setOutput()
+   */
+  public function setOutput(OutputInterface $output) {
+    $this->output = $output;
+
+    return $this;
+  }
+
+  /**
+   *
+   * @return $this
+   *
+   * @see \Robo\Contract\OutputAwareInterface::getOutput()
+   */
+  public function getOutput(): OutputInterface {
+    if (!isset($this->output)) {
+      $this->setOutput(new NullOutput());
+    }
+    return $this->output;
+  }
```

Comment:
Everything appears to have been copied from Robo\IO. I am not convinced this is an optimal implementation.
cc @vishalkhode1

Reply:
@rajeshreeputra It sounds like you're discussing implementing a feature that's similar to something from Robo IO but using our own IO system for testing purposes. That sounds like a practical approach. By implementing it ourself, we have more control over the implementation and can tailor it to fit our specific needs, which could include better integration with our existing systems or customisations to suit our unique requirements. Testing it with our own IO system ensures that it behaves as expected within our environment. Overall, it sounds like a thoughtful approach to building and testing our system

@vishalkhode1 - i think we discussed on this approach as to use required methods and create our own IO as done in this approach

Reply:
Agree with @rajeshreeputra. The idea was here to not copy the complete code from Robo IO, but only the Required methods which will help us in Unit testing of class IO (provided by DRS).